nomachine-client: init at 6.3.6_1

talyz opened the pull request:
Packages mainly the nxplayer part of the client, since the tray
doesn't work very well without the server / a complete installation.

Use the shipped libs, since nxplayer really doesn't like any others. I
believe they use internally modified versions of many libs.

Audio doesn't work: the libasound.so shipped looks for the alsa config
files in the wrong place, and even if it finds them, it still doesn't
work. Using the one from alsaLib doesn't work either and adds
instability.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

talyz committed:
nomachine-client: init at 6.3.6_1

Packages mainly the nxplayer part of the client, since the tray
doesn't work very well without the server / a complete installation.

Use the shipped libs, since nxplayer really doesn't like any others. I
believe they use internally modified versions of many libs.

Audio doesn't work: the libasound.so shipped looks for the alsa config
files in the wrong place, and even if it finds them, it still doesn't
work. Using the one from alsaLib doesn't work either and adds
instability.

talyz commented:
@adisbladis

c0bw3b commented on lib/licenses.nix:
```diff
@@ -498,6 +498,12 @@ lib.mapAttrs (n: v: v // { shortName = n; }) rec {
     fullName  = "University of Illinois/NCSA Open Source License";
   };
 
+  nomachine6 = {
```
It's not worth adding it to the global licenses list since it won't be used by several packages.

Instead you may define the license in-line in the `meta` block of the package definition. [Here is](https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/maxx/default.nix#L82) an example.

c0bw3b commented on pkgs/tools/admin/nomachine-client/default.nix:
```diff
@@ -0,0 +1,90 @@
+      rm $(find . -maxdepth 1 -type f)
+    '';
+  
+    buildPhase = ":";
```
Could be replaced  with `dontBuild = true;`

c0bw3b commented on pkgs/tools/admin/nomachine-client/default.nix:
```diff
@@ -0,0 +1,90 @@
+  
+    buildPhase = ":";
+
+    buildInputs = [ file ];
```
If you only need `file` for the install phase and preFixup hook then it should be in `nativeBuildInputs`

c0bw3b commented on pkgs/tools/admin/nomachine-client/default.nix:
```diff
@@ -0,0 +1,90 @@
+    meta = with stdenv.lib; {
+      description = "NoMachine remote desktop client (nxplayer)";
+      homepage = https://www.nomachine.com/;
+      license = licenses.nomachine6;
```
see my comment above re: license definition

talyz commented:
@c0bw3b Thanks for the review! I have addressed all your comments. I also discovered that nxplayer and nxclient have to be wrapped and have access to some additional files; they were working for me only because the path to my development folder was cached in nomachine's settings.

talyz commented:
Maybe I should have been more clear about this in my last comment: all known issues, except the ones stated in the commit, are now fixed. I've installed it on a second machine with no previous nomachine traces and it works fine there.

c0bw3b commented:
@talyz no problem. It's good to know it's been tested on more than one machine.

c0bw3b commented on pkgs/tools/admin/nomachine-client/default.nix:
```diff
@@ -0,0 +1,97 @@
+    preFixup = ''
+      for i in $out/bin/*; do
+        if file "$i" | grep ELF > /dev/null; then
+          patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
```
Missed it before but `autoPatchelfHook` can probably replaced your whole preFixup hook

See https://nixos.wiki/wiki/Packaging/Binaries#autoPatchelfHook and the example linked from there.

talyz replied:
Right you are. :)

jdanekrh commented:
Thank you for this. Video, mouse and keyboard work fine for me. Sound does not, I reported https://github.com/NixOS/nixpkgs/issues/52507.

talyz commented:
Hi! Yes, you're correct - if you look at the original comment, I left a description of this issue there. I would like to get sound working, but it wasn't a deal-breaker for me: better to have something than nothing at all ;)
